symbolic-debuginfo/src/sourcebundle/utf8_reader.rs

```diff
@@ -0,0 +1,310 @@
+//! UTF-8 reader used by the sourcebundle module to read files.
+
+use std::io::{BufRead, Error, ErrorKind, Read, Result};
+use thiserror::Error;
+
+const MAX_UTF8_SEQUENCE_SIZE: usize = 4;
+const MAX_BUFFER_SIZE: usize = 8 * 1024;
+
+#[derive(Debug, Error)]
+#[error("Invalid UTF-8 sequence")]
+pub struct UTF8ReaderError {
+    /// Make it impossible to construct this struct outside of this module.
+    _a: (),
+}
```

Comment:
You can use `non_exhaustive` to make this nonconstructible outside of the module:
```suggestion
#[non_exhaustive]
pub struct UTF8ReaderError;
```

Reply:
`#[non_exhaustive]` just makes it non-constructible outside the crate, versus what I had which makes it non-constructible outside the module. Honestly, that's probably enough though.

I think the better solution is just to mark this as `pub(crate)`. The only way these errors escape the library is inside a `Box<dyn Error>` (I think wrapped by the SourceBundleError).

Reply:
Having a private field is common practice for this 👍